Sources/GoodSwiftUI/ReadableContentWidthModifier.swift

```diff
@@ -0,0 +1,199 @@
+
+private extension EnvironmentValues {
+
+    @Entry var readableContentInsets = EdgeInsets()
+
+}
+
+// MARK: - Padding modifier
+
+private struct ReadableContentWidthPaddingModifier: ViewModifier {
+
+    @Environment(\.readableContentInsets) private var readableContentInsets
+
+    let alignment: Alignment
+
+    func body(content: Content) -> some View {
+        content
+            .frame(maxWidth: .infinity, alignment: alignment)
+            .padding(.leading, readableContentInsets.leading)
+            .padding(.trailing, readableContentInsets.trailing)
+    }
+
+}
+
+// MARK: - Measurement modifier
+
+private struct ReadableContentWidthMeasurementModifier: ViewModifier {
+
+    @State private var readableContentInsets = EdgeInsets()
+
+    func body(content: Content) -> some View {
+        content
+            .environment(\.readableContentInsets, readableContentInsets)
+            .background(LayoutGuides(onChangeOfReadableContentInsets: {
+                readableContentInsets = $0
+            }))
+    }
+
+}
+
+// MARK: - UIKit LayoutGuides view
+
+private struct LayoutGuides: UIViewRepresentable {
+
+    let onChangeOfReadableContentInsets: (EdgeInsets) -> ()
+
+    func makeUIView(context: Context) -> LayoutGuidesView {
+        let uiView = LayoutGuidesView()
+        uiView.onChangeOfReadableContentInsets = self.onChangeOfReadableContentInsets
+        return uiView
+    }
+
+    func updateUIView(_ uiView: LayoutGuidesView, context: Context) {
+        uiView.onChangeOfReadableContentInsets = self.onChangeOfReadableContentInsets
+    }
+
+}
+
+private final class LayoutGuidesView: UIView {
+
+    fileprivate var onChangeOfReadableContentInsets: (EdgeInsets) -> () = { _ in }
+    private var previousReadableContentInsets: EdgeInsets?
+
+    override func layoutMarginsDidChange() {
+        super.layoutMarginsDidChange()
+        updateReadableContent()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        updateReadableContent()
+    }
+
+    override var frame: CGRect {
+        didSet { updateReadableContent() }
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if traitCollection.layoutDirection != previousTraitCollection?.layoutDirection {
+            updateReadableContent()
+        }
+    }
+
+}
+
+private extension LayoutGuidesView {
+
+    func updateReadableContent() {
+        let isRTLLanguage = traitCollection.layoutDirection == .rightToLeft
+        let readableLayoutFrame = readableContentGuide.layoutFrame
+
+        let readableEdgeInsets = UIEdgeInsets(
+            top: readableLayoutFrame.minY - bounds.minY,
+            left: readableLayoutFrame.minX - bounds.minX,
+            bottom: -(readableLayoutFrame.maxY - bounds.maxY),
+            right: -(readableLayoutFrame.maxX - bounds.maxX)
+        )
+        let readableContentInsets = EdgeInsets(
```

Comment:
Tímpádom pre nejaký jazyk kde sa číta z zprava doľava to nespraví dobre? Len úvaha :) Ale to bol pôvodne dôved prečo to bolo leading a trailing a nie right a left

Reply:
checkujem tam `traitCollection.layoutDirection` presne kvoli tomuto edgecasu 😄 🇮🇷